sources/core/objects/Human.cpp

```diff
@@ -333,21 +337,30 @@ void Human::Live()
         return;
     }
 
+    interface_.UpdateEnviroment();
+
     if (IdPtr<CubeTile> t = owner)
     {
         unsigned int oxygen = t->GetAtmosHolder()->GetGase(OXYGEN);
+        int temperature = t->GetAtmosHolder()->GetTemperature();
+        if(std::abs(40-temperature) > 3)
+        {
+            int damage = (std::abs(40-temperature) / 10) > 1 ? (std::abs(308-temperature) / 100) : 1;
+            burn_damage_ += damage;
+        }
```

Comment:
1. Instead of `std::abs` better to use `qAbs` (because Qt is kind of main library now).
2. Again, constants should be used instead of raw numbers.
3. I feel that it can be done much simpler.
```
const int REGULAR_TEMPERATURE = <some value>;
int temperature_diff = qAbs(REGULAR_TEMPERATURE - temperature);
int damage = qMax(MAX_BURN_DAMAGE, temperature_diff / TEMPERATURE_PER_BURN);
burn_damage_ += damage;
```